hash_practice/exercises.py

```diff
@@ -2,28 +2,224 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+
+        The idea in general is to create hash function, that will return same hashing index  for anagrams.
+
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """

```

Comment:
👍 This is a really sophisticated way to essentially create a hash table of the words, using their letters as a key.  Very neat.